--- /etc/inc/filterold.inc	2010-02-07 14:52:18.602237000 +0800
+++ /etc/inc/filter.inc	2010-02-07 13:00:19.000000000 +0800
@@ -1281,7 +1281,7 @@
 		}
 	}
 
-	if (is_package_installed('squid') && file_exists('/usr/local/pkg/squid.inc')) {
+	if ((is_package_installed('squid') || is_package_installed('lusca-cache')) && file_exists('/usr/local/pkg/squid.inc')) {
 		require_once('squid.inc');
 		$natrules .= squid_generate_rules('nat');
 	}
@@ -2124,7 +2124,7 @@
 	$optcfg = array();
 	generate_optcfg_array($optcfg);
 
-	if (is_package_installed('squid') && file_exists('/usr/local/pkg/squid.inc')) {
+	if ((is_package_installed('squid') || is_package_installed('lusca-cache')) && file_exists('/usr/local/pkg/squid.inc')) {
 		require_once('squid.inc');
 		$ipfrules .= squid_generate_rules('filter');
 	}
--- /etc/inc/system.inc.orig	2010-03-21 20:19:13.627752000 +0800
+++ /etc/inc/system.inc	2010-03-21 20:26:46.000000000 +0800
@@ -1113,6 +1113,8 @@
 function system_halt() {
 	global $g;
 
+	mwexec("/usr/local/etc/rc.d/squid.sh stop");
+
 	system_reboot_cleanup();
 
 	mwexec("nohup /etc/rc.halt > /dev/null 2>&1 &");
@@ -1121,6 +1123,8 @@
 function system_reboot() {
 	global $g;
 
+	mwexec("/usr/local/etc/rc.d/squid.sh stop");
+
 	system_reboot_cleanup();
 
 	mwexec("nohup /etc/rc.reboot > /dev/null 2>&1 &");
@@ -1129,6 +1133,8 @@
 function system_reboot_sync() {
 	global $g;
 
+	mwexec("/usr/local/etc/rc.d/squid.sh stop");
+
 	system_reboot_cleanup();
 
 	mwexec("/etc/rc.reboot > /dev/null 2>&1");
