Index: pkg-utils.inc
===================================================================
--- /etc/inc/pkg-utils.inc	(revision 90)
+++ /etc/inc/pkg-utils.inc	(working copy)
@@ -508,7 +508,7 @@
 	if($pkg_info['after_install_info']) 
 		update_output_window($pkg_info['after_install_info']);
 	update_output_window($static_output);
-	start_service($pkg_info['config_file']);
+	start_service($pkg_info['name']);
 	$restart_sync = true;
 }
 
